plugins/thunderbolt/fu-thunderbolt-controller.c

```diff
@@ -81,22 +77,38 @@ fu_thunderbolt_controller_probe(FuDevice *device, GError **error)
 {
 	FuThunderboltController *self = FU_THUNDERBOLT_CONTROLLER(device);
 	g_autofree gchar *attr_unique_id = NULL;
-	g_autoptr(FuDevice) device_parent = NULL;
+	g_autofree gchar *prop_type = NULL;
 
 	/* FuUdevDevice->probe */
 	if (!FU_DEVICE_CLASS(fu_thunderbolt_controller_parent_class)->probe(device, error))
 		return FALSE;
 
 	/* determine if host controller or not */
-	device_parent =
-	    fu_device_get_backend_parent_with_subsystem(FU_DEVICE(self),
-							"thunderbolt:thunderbolt_domain",
-							NULL);
-	if (device_parent != NULL) {
-		g_autofree gchar *parent_name = g_path_get_basename(
-		    fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device_parent)));
-		if (g_str_has_prefix(parent_name, "domain"))
-			self->controller_kind = FU_THUNDERBOLT_CONTROLLER_KIND_HOST;
+	prop_type = fu_udev_device_read_property(FU_UDEV_DEVICE(self), "USB4_TYPE", NULL);
+	if (prop_type != NULL) {
+		self->controller_kind = fu_thunderbolt_controller_kind_from_string(prop_type);
+	} else {
+		g_autoptr(FuDevice) device_parent =
+		    fu_device_get_backend_parent_with_subsystem(FU_DEVICE(self),
+								"thunderbolt:thunderbolt_domain",
+								NULL);
+		if (device_parent != NULL) {
+			g_autofree gchar *parent_name = g_path_get_basename(
+			    fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device_parent)));
+			if (g_str_has_prefix(parent_name, "domain"))
+				self->controller_kind = FU_THUNDERBOLT_CONTROLLER_KIND_HOST;
+			else
+				self->controller_kind = FU_THUNDERBOLT_CONTROLLER_KIND_DEVICE;
+		}
+	}
+
+	/* hubs have no firmware */
+	if (self->controller_kind == FU_THUNDERBOLT_CONTROLLER_KIND_HUB) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "updates are distributed as part of the platform");
```

Comment:
this doesn't seem right - does your platform natively have a USB4 _hub_ connected to the host controller?  Or is it a downstream hub?

Reply:
The latter

Reply:
Then the error message is wrong, the updates for the downstream hub don't come with the platform.

Reply:
Maybe just something like "hub updates to not accept NVM update payloads"?

Reply:
IIRC Goshen Ridge based devices and ASMedia based devices can be updated from the kernel, but any other vendor's hubs can't. 

What's the actual issue with letting it enumerate?